File: server/entity/command.go
package entity

import "time"

// Command is executed over SSH before or after file transfer
type Command struct {
	ID               uint      `gorm:"primaryKey" json:"id"`
	BackupProfileID  uint      `gorm:"not null;constraint:OnDelete:CASCADE" json:"backup_profile_id"`
	Command          string    `gorm:"not null" json:"command"`
	WorkingDirectory string    `json:"working_directory"`
	RunOrder         int       `gorm:"not null" json:"run_order"`
	RunStage         string    `gorm:"type:text;check:run_stage IN ('pre', 'post')" json:"run_stage"`
	CreatedAt        time.Time `json:"created_at"`
}

File: web/src/types/command.ts
export interface Command {
  id: number;
  backup_profile_id: number;
  command: string;
  working_directory: string;
  run_order: number;
  run_stage: 'pre' | 'post';
  created_at: string;
}

export interface CommandCreateInput {
  command: string;
  working_directory?: string;
  run_order: number;
  run_stage: 'pre' | 'post';
}

export interface CommandUpdateInput {
  command?: string;
  working_directory?: string;
  run_order?: number;
  run_stage?: 'pre' | 'post';
}
